src/compiler/commandLineParser.ts

```diff
@@ -719,6 +761,39 @@ namespace ts {
             wildcardDirectories
         };
 
+        function tryExtendsName(extendedConfig: string): [string[], string[], string[], CompilerOptions] {
+            // If the path isn't a rooted or relative path, don't try to resolve it (we reserve the right to special case module-id like paths in the future)
+            if (!(isRootedDiskPath(extendedConfig) || startsWith(normalizeSlashes(extendedConfig), "./") || startsWith(normalizeSlashes(extendedConfig), "../"))) {
+                errors.push(createCompilerDiagnostic(Diagnostics.The_path_in_an_extends_options_must_be_relative_or_rooted));
+                return;
+            }
+            let extendedConfigPath = toPath(extendedConfig, basePath, getCanonicalFileName);
+            if (!host.fileExists(extendedConfigPath) && !endsWith(extendedConfigPath, ".json")) {

```

Comment:
The reasoning for forbidding modules identifier like paths is, as described in the original issue, to reserve the form for potentially loading configs from `node_modules ` in the future (to align with module resolution) (this way adding that isn't a breaking change) (eslint does this). If you'd like, you can spec that behavior out; but without that I think it'd be better to just reserve the form with an error for now.